rhino/src/main/java/org/mozilla/javascript/NativeGlobal.java

```diff
@@ -131,89 +179,90 @@ public static void init(Context cx, Scriptable scope, boolean sealed) {
-                    }
+    private static void registerGlobalFunction(
+            Scriptable scope, boolean sealed, String name, LambdaFunction fun) {
+        ScriptableObject.defineProperty(scope, name, fun, DONTENUM);
+        if (sealed) {
+            fun.sealObject();
+        }
 
-                case Id_escape:
-                    return js_escape(args);
+        // Strange, but required by the spec and checked by test262
+        fun.setPrototypeProperty(null);
+    }
 
-                case Id_eval:
-                    return js_eval(cx, scope, args);
+    // Eval is special because we need to "recognize" it in isEvalFunction
+    private static void defineGlobalFunctionEval(Scriptable scope, boolean sealed) {
+        LambdaFunction evalFun = new EvalLambdaFunction(scope);
+        registerGlobalFunction(scope, sealed, "eval", evalFun);
+    }
 
-                case Id_isFinite:
-                    {
-                        if (args.length < 1) {
-                            return Boolean.FALSE;
-                        }
-                        return NativeNumber.isFinite(args[0]);
-                    }
+    static boolean isEvalFunction(Object functionObj) {
+        return functionObj instanceof EvalLambdaFunction;
+    }
 
-                case Id_isNaN:
-                    {
-                        // The global method isNaN, as per ECMA-262 15.1.2.6.
-                        boolean result;
-                        if (args.length < 1) {
-                            result = true;
-                        } else {
-                            double d = ScriptRuntime.toNumber(args[0]);
-                            result = Double.isNaN(d);
-                        }
-                        return ScriptRuntime.wrapBoolean(result);
-                    }
+    private static String js_uneval(Context cx, Scriptable scope, Object[] args) {
+        Object value = (args.length != 0) ? args[0] : Undefined.instance;
+        return ScriptRuntime.uneval(cx, scope, value);
+    }
 
-                case Id_isXMLName:
-                    {
-                        Object name = (args.length == 0) ? Undefined.instance : args[0];
-                        XMLLib xmlLib = XMLLib.extractFromScope(scope);
-                        return ScriptRuntime.wrapBoolean(xmlLib.isXMLName(cx, name));
-                    }
+    private static Boolean js_isXMLName(Context cx, Scriptable scope, Object[] args) {
+        Object name = (args.length == 0) ? Undefined.instance : args[0];
+        XMLLib xmlLib = XMLLib.extractFromScope(scope);
+        return ScriptRuntime.wrapBoolean(xmlLib.isXMLName(cx, name));
+    }
 
-                case Id_parseFloat:
-                    return js_parseFloat(args);
+    private static Boolean js_isNaN(Object[] args) {
+        // The global method isNaN, as per ECMA-262 15.1.2.6.
+        boolean result;
+        if (args.length < 1) {
```

Comment:
Why not using the same coding style as below and directly return here?

    return Boolean.TRUE;